packages/react-hooks/src/ElementUtils.ts

```diff
@@ -7,10 +12,13 @@ import { InferComponentProps } from '@deephaven/utils';
  * @param type The type to check against
  * @returns True if the node is a React element of the specified type
  */
-export function isElementOfType<T>(
-  node: ReactNode,
-  type: T
-): node is ReactElement<InferComponentProps<T>> {
+export function isElementOfType<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  T extends string | JSXElementConstructor<any> =
```

Comment:
Just curious for learning purposes, what is the reasoning for having to separately extend string here, not any other types?

Reply:
Element types are either `string` or a factory function. e.g. `'div' | 'span' | SomeComponent`. This also matches the signature of the `ReactElement` 2nd generic arg